src/pricecypher/datasets.py

```diff
@@ -0,0 +1,396 @@
+
+        # Make sure all values are strings.
+        values = list(map(str, values))
+        # Filter and create new collection
+        scope_values = [sv for sv in self._list if sv.value in values]
+        return ScopeValueCollection(scope_values)
+
+    def pluck(self, prop):
+        """
+        Get a list with the values of the given prop.
+
+        :param str prop: Property to pluck from the scope values.
+        :return:
+        """
+        return [getattr(v, prop) for v in self._list]
+
+
+class Datasets(object):
+    """
+    Datasets API. Exposes all available operations on datasets, like fetching the available datasets, listing
+    the scopes within a dataset, retrieving transactions within a dataset, etc.
+
+    :param str bearer_token: Bearer token for PriceCypher (logical) API. Needs 'read:datasets' scope.
+    :param str users_base: (optional) Base URL for PriceCypher user tool API. Used to find base URL of dataset service,
+        when no dss_base is provided.
+        (defaults to https://users.pricecypher.com)
+    :param str dss_base: (optional) Base URL for PriceCypher dataset service API.
+        (defaults to dss_url property as returned for the dataset by the PriceCypher user tool API)
+    :param RestClientOptions rest_options: (optional) Set any additional options for the REST client, e.g. rate-limit.
+        (defaults to None)
+    """
+
+    def __init__(self, bearer_token, users_base='https://users.pricecypher.com', dss_base=None, rest_options=None):
+        self._bearer = bearer_token
+        self._users_base = users_base
+        self._dss_base = dss_base
+        self._rest_options = rest_options
+        self._client = RestClient(jwt=bearer_token, options=rest_options)
+        self._all_meta = None
+
+    def _ds_base(self, dataset_id):
+        """
+        Get dataset service url base for the given dataset ID.
+        Will be fetched from dataset META if no dss_base present.
+
+        :param int dataset_id: Dataset ID to get base URL for.
+        """
+        if self._dss_base is not None:
+            return '{}/datasets/{}'.format(self._dss_base, dataset_id)
+
+        return self.get_meta(dataset_id).dss_url
+
+    def index(self):
+        """
+        List all available datasets the user has access to.
+        Response is cached in this instance for as long as this instance lives.
+
+        :return: list of datasets.
+        :rtype list[Dataset]
+        """
+        if self._all_meta is not None:
+            return self._all_meta
+
+        self._all_meta = UsersEndpoint(self._bearer, self._users_base, self._rest_options).datasets().index()
+        return self._all_meta
+
+    def get_meta(self, dataset_id):
+        """
+        Get metadata like the dataset service url and time of creation of a dataset
+
+        :param dataset_id: Dataset to get metadata for.
+        :rtype: Dataset
+        """
+        return next((d for d in self.index() if d.id == dataset_id), None)
```

Comment:
```suggestion
        return next([d for d in self.index() if d.id == dataset_id], None)
```
Personal preference, I guess, but I like using `[` for notation of list comprehension to make clearer that it results in an array/list.

Reply:
Applies to other parts of this PR as well. Haven't commented on them all, but generally can be found by `((`.

Reply:
I don't really see why using list comprehension would be preferred in this case. In general, I believe using generators in Python is a bit more efficient since with list comprehension memory is reserved for the entire list, whereas with generators the items are generated only when needed. Since we are only interested in the first item, it makes more sense to me to use a generator in expressions like this. In this case, it wouldn't really matter though, since the list would only be of size 1. Still, I don't see a reason why we should use list comprehension over a generator here.

Reply:
I had no clue that this made a difference between creating a generator and performing list comprehension, I thought it was just different syntax for the same thing. Then generators are preferred for sure, yes!